apps/dashboard/src/components/subscribers/subscriber-list.tsx

```diff
@@ -0,0 +1,129 @@
+import { CursorPagination } from '@/components/cursor-pagination';
+import { Table, TableBody, TableHead, TableHeader, TableRow } from '@/components/primitives/table';
+import { SubscriberListEmpty } from '@/components/subscribers/subscriber-list-empty';
+import { SubscriberRow, SubscriberRowSkeleton } from '@/components/subscribers/subscriber-row';
+import { defaultSubscribersFilters, SubscribersFilters } from '@/components/subscribers/subscribers-filters';
+import { useFetchSubscribers } from '@/hooks/use-fetch-subscribers';
+import { useSubscribersUrlState } from '@/hooks/use-subscribers-url-state';
+import { cn } from '@/utils/ui';
+import { HTMLAttributes, useState } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+type SubscriberListProps = HTMLAttributes<HTMLDivElement>;
+
+export const SubscriberList = (props: SubscriberListProps) => {
+  const { className, ...rest } = props;
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
+  const [cursorHistory, setCursorHistory] = useState<string[]>(['']);
+
+  const { filterValues, handleFiltersChange } = useSubscribersUrlState();
+  const email = searchParams.get('email') || '';
+  const phone = searchParams.get('phone') || '';
+  const name = searchParams.get('name') || '';
+  const subscriberId = searchParams.get('subscriberId') || '';
+  const limit = parseInt(searchParams.get('limit') || '10');
+  const currentCursor = searchParams.get('cursor') || '';
+
+  const currentIndex = cursorHistory.indexOf(currentCursor);
+
+  const { data, isPending, isError } = useFetchSubscribers({
+    cursor: currentCursor,
+    limit,
+    email,
+    phone,
+    subscriberId,
+    name,
+  });
+
+  const handleNext = () => {
+    if (!data?.nextCursor) return;
+
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set('cursor', data.nextCursor);
+
+    // Use replace instead of navigate to avoid history stack issues
+    navigate(`${location.pathname}?${newParams}`, { replace: true });
```

Comment:
but if we store everything in the URL I think it's important to support browser back/forward behavior